internal/experiment/trial_template.go

```diff
@@ -59,4 +60,31 @@ func PopulateTrialFromTemplate(exp *optimizev1beta2.Experiment, t *optimizev1bet
 	if t.Namespace == "" && exp.Spec.NamespaceSelector == nil && exp.Spec.NamespaceTemplate == nil {
 		t.Namespace = exp.Namespace
 	}
+
+	// Constant parameters are only part of the experiment, so they must be added here
+	for _, p := range exp.Spec.Parameters {
+		if v := ParameterConstant(p); v != nil {
+			t.Spec.Assignments = append(t.Spec.Assignments, optimizev1beta2.Assignment{
+				Name:  p.Name,
+				Value: *v,
+			})
+		}
+	}
+}
+
+// ParameterConstant returns the value of the supplied parameter if it is constant.
+// A constant parameter has a domain which includes exactly 1 value (i.e. min==max
+// for numeric, or len(values)==1 for categorical); these parameters are not processed
+// on the server.
+func ParameterConstant(p optimizev1beta2.Parameter) *intstr.IntOrString {
+	switch {
+	case p.Min != p.Max || len(p.Values) > 1:
+		return nil
```

Comment:
nit, I'd switch around this case with the default so we highlight the actual cases we care about, 

```go
switch {
case p.Min == p.Max && len(p.Values) == 0:
		v := intstr.FromInt(int(p.Max))
 		return &v
case len(p.Values) == 1:
		v := intstr.FromString(p.Values[0])
 		return &v
default:
		return nil
}
```